README.md

```diff
@@ -101,6 +101,35 @@ Finally, run `nix-build`, and voilà, in `./result/` you find the project with
 all its dependencies correctly linked to their corresponding `node_modules`
 folder, recursively.
 
+## Using private package repository
+
+Since `yarn2nix` uses standard `fetchurl` to download packages,
+it is possible to authenticate by overriding `fetchurl` to use
+`netrc` file.
+
+```nix
+let
+  auth = { login: "your_login"; taken "your_auth_token"; };
+  config = {
+    packageOverrides = pkgs: rec {
+      fetchurl = otps: pkgs.fetchurl (opts // {
+        netrcPhase = ''
+          cat << EOF > /build/netrc
+            machine repository_url
+              login ${auth.login}
+              password ${auth.token}
+EOF
+        '';
+      });
+    };
+  };
+  nixpkgsPath = <nixpkgs>;
+  pkgs = import nixpkgsPath { inherit config };
+  ...
+```
+
+Consider this [wiki](https://nixos.wiki/wiki/Enterprise) article for more help.
```

Comment:
Shouldn’t we follow the guide in the wiki more closely here, e.g. the user puts the credentials in his `/etc/netrc` and then just overrides `fetchurl` with `fetchurlBoot`?

Reply:
probably, will play with this approach in our code and see if it works nicely

Reply:
the problem with `fetchurlBoot` is that it [requires](https://github.com/NixOS/nixpkgs/blob/ceee97c038c9b0ec48409d3b3313cb77c2543309/pkgs/build-support/fetchurl/boot.nix) `sha256` and npm gives out sha1 and sha512. Do you have any idea haw I could move this forward? @Profpatsch

Reply:
Patch it to accept either? `<nix/fetchurl.nix>` can accept all these hashes (`nix-instantiate --eval -E '<nix/fetchurl.nix>'` and open the file name it outputs).